app/src/main/cpp/emu_jni.cpp

```diff
@@ -7,35 +7,25 @@
 #include <android/log.h>
 #include "skyline/loader/loader.h"
 #include "skyline/common.h"
+#include "skyline/common/settings.h"
 #include "skyline/os.h"
 #include "skyline/jvm.h"
+#include "skyline/gpu.h"
 #include "skyline/input.h"
+#include "skyline/kernel/types/KProcess.h"
 
-bool Halt;
-jobject Surface;
-skyline::GroupMutex JniMtx;
-skyline::u16 fps;
-skyline::u32 frametime;
-std::weak_ptr<skyline::input::Input> inputWeak;
-
-void signalHandler(int signal) {
-    __android_log_print(ANDROID_LOG_FATAL, "emu-cpp", "Halting program due to signal: %s", strsignal(signal));
-    exit(signal);
-}
+skyline::u16 Fps;
+skyline::u32 FrameTime;
+std::weak_ptr<skyline::kernel::OS> OsWeak;
+std::weak_ptr<skyline::gpu::GPU> GpuWeak;
+std::weak_ptr<skyline::input::Input> InputWeak;
 
 extern "C" JNIEXPORT void Java_emu_skyline_EmulationActivity_executeApplication(JNIEnv *env, jobject instance, jstring romUriJstring, jint romType, jint romFd, jint preferenceFd, jstring appFilesPathJstring) {
-    Halt = false;
-    fps = 0;
-    frametime = 0;
-
-    std::signal(SIGTERM, signalHandler);
-    std::signal(SIGSEGV, signalHandler);
-    std::signal(SIGINT, signalHandler);
-    std::signal(SIGILL, signalHandler);
-    std::signal(SIGABRT, signalHandler);
-    std::signal(SIGFPE, signalHandler);
+    Fps = 0;
+    FrameTime = 0;
 
-    setpriority(PRIO_PROCESS, static_cast<id_t>(gettid()), -8); // Set the priority of this process to the highest value
+    pthread_setname_np(pthread_self(), "EmuMain");
+    setpriority(PRIO_PGRP, static_cast<id_t>(gettid()), -8); // Set the priority of this process to the highest value
```

Comment:
This is for the entire Skyline emulator process group, I've removed the guest -> host priority thing as I think it's pointless because the same semantics don't apply.

Reply:
don't we still have rescaling stuff?

But yeah I would keep it at default then, setting high priority unnecessarily starves other processes

Reply:
This is recommended behaviour for the foreground application (Android will automatically reset this when it's not in focus iirc), I would also set the core mask to the high performance cores but we'd want our worker threads to potentially be scheduled on small cores.